reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/RumConfigurationTest.kt

```diff
@@ -0,0 +1,291 @@
+        val expectedEventsCount = expectedSessionCount * 2 // 2 events per view, i.e. Start & Stop
+        val expectedEventsOffset = expectedSessionOffset * 2 // 2 events per view, i.e. Start & Stop
+
+        // When
+        repeat(repeatCount) {
+            rumMonitor.startView(viewKey, viewName, emptyMap())
+            rumMonitor.stopView(viewKey, emptyMap())
+            rumMonitor.stopSession()
+        }
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        assertThat(eventsWritten.size).isCloseTo(expectedEventsCount, offset(expectedEventsOffset))
+    }
+
+    @RepeatedTest(16)
+    fun `M send all events W samplingRate(100)`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String
+    ) {
+        // Given
+        val fakeRumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .setSessionSampleRate(100f)
+            .build()
+        Rum.enable(fakeRumConfiguration, stubSdkCore)
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+        val repeatCount = 64
+        val expectedEventsCount = repeatCount * 2 // 2 events per view, i.e. Start & Stop
+
+        // When
+        repeat(repeatCount) {
+            rumMonitor.startView(viewKey, viewName, emptyMap())
+            rumMonitor.stopView(viewKey, emptyMap())
+            rumMonitor.stopSession()
+        }
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        assertThat(eventsWritten.size).isEqualTo(expectedEventsCount)
+    }
+
+    @RepeatedTest(16)
+    fun `M send mapped events W setResourceEventMapper()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @StringForgery originalResourceUrl: String,
+        @StringForgery mappedResourceUrl: String,
+        @Forgery resourceMethod: RumResourceMethod,
+        @IntForgery(200, 599) statusCode: Int,
+        @LongForgery size: Long,
+        @Forgery resourceKind: RumResourceKind
+    ) {
+        // Given
+        val fakeRumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .setResourceEventMapper {
+                it.apply {
+                    resource.url = mappedResourceUrl
+                }
+            }
+            .build()
+        Rum.enable(fakeRumConfiguration, stubSdkCore)
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.startResource(originalResourceUrl, resourceMethod, originalResourceUrl, emptyMap())
+        rumMonitor.stopResource(originalResourceUrl, statusCode, size, resourceKind, emptyMap())
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        StubEventsAssert.assertThat(eventsWritten)
+            .hasRumEvent(1) {
```

Comment:
minor: maybe we need to make use of the named argument here, otherwise it is hard to understand what `1` means here without looking at the assertion method implementation details.